pkgs/top-level/module/tests.nix

```
@@ -0,0 +1,115 @@
+/*
+  Run
+  [nixpkgs]$ nix-build -A tests.top-level.module --show-trace
+
+  Repl
+  [nixpkgs]$ nix repl pkgs/top-level/module/test.nix --show-trace
+  nix-repl> tests
+
+  Maintainer: @roberth
+*/
+
+# A the tests are invoked using an already invoked nixpkgs, we're going to
+# mostly ignore this pkgs.
+# To catch silly mistakes, let's rename this to pkgs_.
+{ pkgs }:
+let pkgs_ = pkgs;
+in
+let pkgs = throw "You probably didn't mean to use pkgs, but rather use a more specific pkgs variable in this test.";
+
+    inherit (pkgs_) lib emptyFile;
+    inherit (lib) evalModules;
+in
```

Now that the code is written, we know what is needed :)
```suggestion
{ lib, emptyFile }:
```